code/common/window_handle.h

```diff
@@ -1,17 +1,34 @@
 #pragma once
-#include <Windows.h>
+#if defined(_WIN32)
+    #include <Windows.h>
+#elif defined(__linux__)
+
+#else
+    static_assert(false, "Not implemented");
+#endif
+
 #include "common/platform.h"
 
+
 namespace common
 {
   template<enum platform>
   struct window_handle_t;
 
+  #if defined(_WIN32)
   template<>
   struct window_handle_t<platform::windows>
   {
     HWND window_handle{ nullptr };
   };
+  #elif defined(__linux__)
+  template<>
+  struct window_handle_t<platform::linuxplatform>
+  {
+    // Don't know the type to go in here yet
+    //HWND window_handle{ nullptr };
```

Comment:
For linux you want
```cpp
void* connection; // Display*
void* handle;     // Window
```
For now.

Not ideal but being type safe here on linux is extra painful not just because of the two different window server apis but also because x11 likes to define a bunch of junk like `none` which will cause issues in the rest of the codebase so it's best to limit it's inclusion to just the cxx file that needs it.

I guess another option is to forward declare.

Reply:
Addressed in 02966ccf343cdd5a0222c38d92fd379eaaf40d6d.
Can reconsider implementation when we start looking at `host`?